pubtools/pulplib/_impl/client/client.py

```diff
@@ -181,10 +184,52 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def get_maintenance_report(self):
+        """Get maintenance report from maintenance repository
+
+        Returns:
+            Future[:class:`~pubtools.pulplib.MaintenceReport`]
+                A future describes the maintenance status
+        """
+        report_ft = self._do_get_maintenance()
+
+        return f_map(
+            report_ft,
+            lambda data: MaintenanceReport.from_data(data)
+            if data
+            else MaintenanceReport(),
+        )
+
+    def set_maintenance(self, report):
+        """Set repositories to maintenance mode.
+
+        Args:
+            report:
+                An updated :class:`~pubtools.pulplib.MaintenanceReport` object that
+                will be used as the newest maintenance report.
+        Return:
+            Future[:class:`~pubtools.pulplib.Task`]
+                A future which is resolved when publish succeeds.
```

Comment:
Again I would rather be a bit more vague and generic with respect to what's actually happening here.  No need to commit to a particular implementation here - we might need to add or remove steps to maintenance mode update in the future.

So, I think we shouldn't mention here about "maintenance repository" or "publish", e.g. :

```suggestion
                A future which is resolved when maintenance mode has been updated successfully.
                
                The future returns a list of zero or more Pulp tasks created during the update.
```

I see the main purpose of returning anything is just to allow for log messages matching up with Pulp tasks, such as:

```
Updated maintenance mode: <task1>, <task2>
```

...so that it's possible to look at Pulp task logs and find the matching pubtools-pulp log, and vice-versa.  The caller ideally won't depend on returned tasks being specifically publish tasks.